proto/dwtools/abase/l1.test/Str.test.s

```diff
@@ -8835,6 +8835,99 @@ ghij
 
 //
 
+function strLinesNearestReport( test )
+{
+  test.case = 'one letter to the left';
+  var src = 'function add( x,y ) { return x + y }';
+  var got = _.strLinesNearestReport({ src, charsRangeLeft : [ 1, 7 ], gray : 1, numberOfLines : 1 });
+  var expected =  [ 'f', 'unctio', 'n add( x,y ) { return x + y }' ];
+  test.identical( got.nearest, expected );
+
+  test.case = 'a few letters to the left';
+  var src = 'function add( x,y ) { return x + y }';
+  var got = _.strLinesNearestReport({ src, charsRangeLeft : [ 5, 7 ], gray : 1, numberOfLines : 1 });
```

Comment:
Please, add test cases with ranges like 
```js
[ 5, 5 ];
[ 5, 3 ];
[ -1, 2];
[ 5, -1 ];
```
Some of them can throws error, I'm not sure of it. 
Maybe, some behavior of the routine should be improved with this test cases.